grr/server/grr_response_server/gui/api_plugins/metadata_test.py

```diff
@@ -0,0 +1,382 @@
+        .get("schema")
+    )
+
+    expected = dict()
+
+    expected_double = {"type": "number", "format": "double"}
+    expected["field_double"] = expected_double
+
+    expected_float = {"type": "number", "format": "float"}
+    expected["field_float"] = expected_float
+
+    expected_int64 = {"type": "integer", "format": "int64"}
+    expected["field_int64"] = expected_int64
+
+    expected_uint64 = {"type": "integer", "format": "uint64"}
+    expected["field_uint64"] = expected_uint64
+
+    expected_int32 = {"type": "integer", "format": "int32"}
+    expected["field_int32"] = expected_int32
+
+    expected_fixed64 = {"type": "integer", "format": "uint64"}
+    expected["field_fixed64"] = expected_fixed64
+
+    expected_fixed32 = {"type": "integer", "format": "uint32"}
+    expected["field_fixed32"] = expected_fixed32
+
+    expected_bool = {"type": "boolean"}
+    expected["field_bool"] = expected_bool
+
+    expected_string = {"type": "string"}
+    expected["field_string"] = expected_string
+
+    expected_bytes = {"type": "string", "format": "binary"}
+    expected["field_bytes"] = expected_bytes
+
+    expected_uint32 = {"type": "integer", "format": "uint32"}
+    expected["field_uint32"] = expected_uint32
+
+    expected_sfixed32 = {"type": "integer", "format": "int32"}
+    expected["field_sfixed32"] = expected_sfixed32
+
+    expected_sfixed64 = {"type": "integer", "format": "int64"}
+    expected["field_sfixed64"] = expected_sfixed64
+
+    expected_sint32 = {"type": "integer", "format": "int32"}
+    expected["field_sint32"] = expected_sint32
+
+    expected_sint64 = {"type": "integer", "format": "int64"}
+    expected["field_sint64"] = expected_sint64
+
+    expected_binarystream = {"type": "string", "format": "binary"}
+    expected["BinaryStream"] = expected_binarystream
+
+    self.assertDictEqual(expected, openapi_primitives)
+
+  def testRepetitiveField(self):
+    # Test the GET route. The repeated field should be a query parameter.
+    params = (self.openapi_desc_dict.get("paths").get("/metadata_test/method4")
+              .get("get").get("parameters"))
+    self.assertEqual(len(params), 1)
+
+    param = params[0]
+    self.assertEqual(param["in"], "query")
+    schema = param["schema"]
+    self.assertEqual(schema["type"], "array")
+    self.assertEqual(schema["items"]["type"], "integer")
+    self.assertEqual(schema["items"]["format"], "int64")
+
+    # Test the POST route. The repeated field should be in the request body.
+    req_body = (self.openapi_desc_dict.get("paths")
+                .get("/metadata_test/method4").get("post").get("requestBody"))
+    req_schema = req_body["content"]["application/json"]["schema"]
+    self.assertEqual(req_schema["type"], "object")
+    schema = req_schema["properties"]["field_repeated"]
+    self.assertEqual(schema["type"], "array")
+    self.assertEqual(schema["items"]["type"], "integer")
+    self.assertEqual(schema["items"]["format"], "int64")
```

Comment:
Same question here: can we compare the whole `schema` object?

Reply:
Made the change, thank you!